packages/itwinui-react/src/core/List/ListItem.tsx

```diff
@@ -0,0 +1,107 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import cx from 'classnames';
+import { useTheme } from '../utils';
+import type {
+  PolymorphicForwardRefComponent,
+  PolymorphicComponentProps,
+} from '../utils';
+import '@itwin/itwinui-css/css/menu.css';
+
+const ListItemComponent = React.forwardRef((props, ref) => {
+  const {
+    as: Element = 'li',
+    size = 'default',
+    disabled = false,
+    active = false,
+    actionable = false,
+    focused = false,
+    className,
+    ...rest
+  } = props;
+
+  useTheme();
+
+  return (
+    <Element
+      className={cx('iui-list-item', className)}
+      data-iui-active={active ? 'true' : undefined}
+      data-iui-disabled={disabled ? 'true' : undefined}
+      data-iui-size={size === 'large' ? 'large' : undefined}
+      data-iui-actionable={actionable ? 'true' : undefined}
+      data-iui-focused={focused ? 'true' : undefined}
```

Comment:
Does users need to add their own focus functionality?

Reply:
This is for custom focus management (like we do in Select/Combobox).

In the css I have other selectors too, so if the listitem is focusable or contains a LinkAction, then it will have focus styling.
![](https://user-images.githubusercontent.com/9084735/230392927-c517517b-ccf8-463c-b0e7-47ec1f74a72c.png)

Reply:
Should actionable list items be automatically focusable?

Reply:
What does "automatically focusable" mean?

Reply:
Right now I can't tab through actionable list items. Or did you do this on purpose for users to add all functionality themselves? (Comparing to MenuItem)

Reply:
We can't simply add `tabindex` as it is not enough by itself. Listitems are generally meant for displaying data rather than for interaction. When it needs to be interactable, that usually happens through either a child element (e.g. a link) or entirely custom focus and role management (e.g. MenuItem has `role=option`, roving tabindex, and arrow key handling).

In a way, you can look at `ListItem` as a low-level primitive for more advanced interactive cases. I expect we will use it inside `Listbox` and in the future also inside `Menu`/`Select`/`ComboBox`.

Maybe the name "actionable" is misleading then. But I did mention in the jsdoc that it is only for styling. And I will have more details in the documentation page as well.
![](https://user-images.githubusercontent.com/9084735/231166852-3227b57a-4bf0-4826-8097-b47af4c22a41.png)